quinn/append_if_schema_identical.py

```diff
@@ -0,0 +1,33 @@
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType, StructField, IntegerType, StringType, DoubleType
+
+
+class SchemaMismatchError(ValueError):
+    """raise this when there's a schema mismatch between source & target schema"""
+
+
+def append_if_schema_identical(source_df: DataFrame, target_df: DataFrame):
+    # Retrieve the schemas of the source and target dataframes
+    source_schema = source_df.schema
+    target_schema = target_df.schema
+
+    # Convert the schemas to a list of tuples
+    source_schema_list = [(field.name, str(field.dataType)) for field in source_schema]
+    target_schema_list = [(field.name, str(field.dataType)) for field in target_schema]
+
+    unmatched_cols = [col for col in source_schema_list if col not in target_schema_list]
+
+    print(source_schema_list)
+    print(type(source_schema_list[0][0]))
+    print(type(target_schema_list[0][0]))
+    print(target_schema_list)
```

Comment:
Such prints may be useful for debugging but it is bad idea to print something from production. Please delete this prints or add an optional argument to the function (for example, `verbose:bool=False`) and print anything in verbose-mode only.

Reply:
@SemyonSinchenko : I will remove the comments as suggested.